File: 1.two-sum/c++/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;        
        unordered_map<int, int> list;
        
        for (int i = 0; i < nums.size(); i++) {
            if (list.find(target - nums[i]) != list.end()) {
                ret.push_back(list[target - nums[i]]);
                ret.push_back(i);
            } else {
                list[nums[i]] = i;
            }
        }
        return ret;
    }
};

File: 1.two-sum/py/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        return [i for i in range(len(nums)) if target-nums[i] in nums[:i]+nums[i+1:]]

